sdk/attestation/azure-security-attestation/test/samples/attestation/attestation.cpp

```diff
@@ -0,0 +1,151 @@
+    AttestOptions attestOptions;
+    attestOptions.RuntimeData
+        = AttestationData{AttestationCollateral::RuntimeData(), AttestationDataType::Json};
+
+    Azure::Response<AttestationToken<AttestationResult>> sgxResult
+        = attestationClient.AttestSgxEnclave(sgxEnclaveQuote, attestOptions);
+
+    std::cout << "SGX Quote MRSIGNER is: "
+              << Azure::Core::Convert::Base64Encode(sgxResult.Value.Body.SgxMrSigner.Value())
+              << std::endl;
+    std::cout << "SGX Quote MRENCLAVE is: "
+              << Azure::Core::Convert::Base64Encode(sgxResult.Value.Body.SgxMrEnclave.Value())
+              << std::endl;
+
+    std::cout << "Attestation Token runtimeData is " << sgxResult.Value.Body.RuntimeClaims.Value()
+              << std::endl;
+  }
+
+  /** @brief Simple call to attestation specifying a predefined SGX quote.
+   *
+   * Note that calls to AttestSgxEnclave are anonymous and do not require authentication.
+   */
+  void SampleAttestSgxEnclaveWithBinaryRuntimeData()
+  {
+    std::cout << "In function: SampleAttestSgxEnclaveWithBinaryRuntimeData" << std::endl;
+    AttestationClientOptions clientOptions;
+    // create client
+    AttestationClient attestationClient(std::getenv("ATTESTATION_ISOLATED_URL"), clientOptions);
+
+    std::vector<uint8_t> sgxEnclaveQuote = AttestationCollateral::SgxQuote();
+
+    // Set the RuntimeData in the request to the service. Ask the service to interpret the
+    // RuntimeData as a JSON object when it is returned in the resulting token.
+    AttestOptions attestOptions;
+    attestOptions.RuntimeData
+        = AttestationData{AttestationCollateral::RuntimeData(), AttestationDataType::Binary};
+
+    Azure::Response<AttestationToken<AttestationResult>> sgxResult
+        = attestationClient.AttestSgxEnclave(sgxEnclaveQuote, attestOptions);
+
+    std::cout << "SGX Quote MRSIGNER is: "
+              << Azure::Core::Convert::Base64Encode(sgxResult.Value.Body.SgxMrSigner.Value())
+              << std::endl;
+    std::cout << "SGX Quote MRENCLAVE is: "
+              << Azure::Core::Convert::Base64Encode(sgxResult.Value.Body.SgxMrEnclave.Value())
+              << std::endl;
+
+    std::cout << "Attestation Token runtimeData is "
+              << Azure::Core::Convert::Base64Encode(sgxResult.Value.Body.EnclaveHeldData.Value())
+              << std::endl;
+  }
+
+}}}} // namespace Azure::Security::Attestation::Samples
+
+int main()
+{
+  try
+  {
+    Azure::Security::Attestation::Samples::SampleAttestSgxEnclaveSimple();
+    Azure::Security::Attestation::Samples::SampleAttestSgxEnclaveWithJSONRuntimeData();
+  }
+  catch (Azure::Core::Credentials::AuthenticationException const& e)
+  {
+    std::cout << "Authentication Exception happened:" << std::endl << e.what() << std::endl;
+    return 1;
+  }
+  catch (Azure::Core::RequestFailedException const& e)
+  {
+    std::cout << "Attestation Client Exception happened:" << std::endl << e.Message << std::endl;
```

Comment:
Can we use `e.what()` here as the lines above?  or we have different stuff for each?

Reply:
This was modeled from the keyvault samples, and that's what they did. I can absolutely change it however.

Reply:
I am not sure if there's a `e.what()` on `RequestFailException`.  or what it give us different than the e.message

Reply:
There should be, `RequestFailedException` derives from std::exception, which has a `.what` method. However it appears that `RequestFailedException` doesn't appear to pass the `Message` to the ctor (which makes sense based on how that works).